### ISSUE TITLE
Antelope-mspass interaction with the net/sta/chan/loc SEED problem

### ISSUE THREAD

**Issue by pavlis** [wangyinz/mspass#57]
As you know there is a nasty problem in mapping between Antelope's database and they way SEED is treated by obspy and data centers.  All the later index all channels as the combination of five keys:   net, sta, chan, loc, and a time interval.   This makes for a very messy situation that is made worse by the fact the old CSS3.0 standard had no concept of net or loc codes.   BRTT in Antelope developed kludge that never really worked right but which they have blundered through for many years now.  They use the nonstandard relational tables snetsta and chanloc (or something like that) that creates (only when needed) keys like IU_AAK for net=IU and sta=AAK.   Similarly loc can create weird chan codes like BHZ_01 and BHZ_00.   That mess has influenced our design by aiming to not depend on the (data center appropriate) concept of net/sta/chan/loc. 

This issue will rear it's head only when trying to export to antelope or importing data assembled with antelope.  It comes up in the short term because I've been assembling usarray data through obspy's web service mass downloader and will face this issue soon.  The reason is I am planning a bit of a mixed obspy/antelope method for building a working 3C dataset.  I'm going to use obspy to assemble site and source collection data using the new tools recently developed.  For the waveforms, however, I want to use the extract_events program I wrote years ago.  It has useful features to assure data are resampled onto a common sample rate and assembled the data into a workable relative time base (origin time as 0 or arrival time as 0).  Thus, I'm going to face the issue of conflicting net/sta and/or conflicting chan/loc.   

With that ridiculously long introduction the question is a design for handling import and export from a css3.0 database in general and antelope in particular.   

I am thinking the best way to handle this will be to demand the first step is to assemble a full obspy Inventory object that characterizes the dataset.  It will be easy to scan that data to build a list of potentially conficting net/sta and chan/loc combinations.   Whether build dynamically or loaded from a file that pair of lists is the starting point for resolving conflicts.   After I think we need only supply some basic tools for sorting this out and state the user has to deal with his own data problems.  We can supply the things I will be forced to develop for these test data as a starting point.   

What do you think?

**Comment by wangyinz** [comment]
I agree with this approach. Also, I think there are more obspy users than antelope users in the community, and our tool is already heavily based on obspy, so it should be sufficient to provide the basic tools to sort out the conflicting keys. These tool will be mostly for us instead of the majority of the users of mspass.

**Comment by wangyinz** [comment]
Closing this as completed/superseded by the current receiver-metadata model.

MsPASS now models StationXML/SEED metadata with separate `site` and `channel` collections, time intervals, and MongoDB ObjectId cross-references. The database documentation defines the net/sta/loc and net/sta/chan/loc lookup conventions, while Inventory import and normalization code handle those records.

A specific Antelope import/export collision can be filed with a minimal dataset and the current importer involved.
